### DISCUSSION TITLE
localhost has working POST and PUT, vercel has errors

### DISCUSSION THREAD

**Discussion by t0bbl** [neuefische/questions#615] [Web] [answered]
## 🎉 Introduction:

My page is perfectly working on my local machine, as soon as i deploy it to vercel the POST and PUT requests give different errors.

## 🧮 Pull Request or link to code:

[PR](https://github.com/t0bbl/capstone-project/pull/34)

<img width="991" alt="Screenshot 2023-05-23 at 22 35 57" src="https://github.com/neuefische/questions/assets/117731660/b899c8cc-19ed-4b45-87c1-73a484c4d1ca">

## ♻️ What are the steps to reproduce this issue?

1. put three Keywords in the Form to send a POST that is working
2. wait till its loaded
3. choose one of the four pictures provided by dall E 
4. try to favorite / unfavorite the picture
5. the network console shows the error messages

the exact same procedure run locally works perfect!

## 🤔 What happens?

- the three keywords are saved to the db,
- the post to openai is made
- the response is displayed and the url of all pictures is added to the db entry
- you click on one of the pictures
- you click favorite and its saved to cloudinary but the saving to the DB gets error405 and every click after that 400 for trying to update the DB

## 👉 What did you expect to happen?

- the three keywords are saved to the db,
- the post to openai is made
- the response is displayed and the url of all pictures is added to the db entry
- you click on one of the pictures
- you click favorite and its saved to cloudinary, saved to another collection in the same DB and to your localstorage (this was kind of a placeholder for the implementation of user logins and profils)
<img width="251" alt="Screenshot 2023-05-23 at 22 41 28" src="https://github.com/neuefische/questions/assets/117731660/2b3acd56-6a3c-427e-baa8-8a7c3708fc60">
-  you click unfavorite and it gets unfavorited in your localstorage and the rating gets decreased in the DB
- you click favorite again and it gets favorited in your localstorage once more and the rating is increased.

## 🤖 Any logs, error output, etc?

see screenshots above.

## ✍️ Any other comments?

as pointed out it works perfectly in localstorage. 
i double checked my environment variables on vercel (its for aimyshirtup, as you can see in the screenshot above. i dont need any extra login for different collections do i?). 
i tried different file structures/names/folderstructures for my api endpoints.
 i checked the schema but it should be good for what iam doing.

Thanks for having a look.

**Comment by t0bbl** [comment]
i just became aware that the allfavorites pages isnt working either.
it must be a problem with the Favorites collection!?

**Comment by NikRoe** [comment, marked as answer]
Hi Robin,

in your `/pages/AlltimeFavorites` you use useSWR to fetch one of your API Endpoints. However, the capitalization does not match.

Your route is called `/api/Favorites/alltimeFavorites/allFavorites`, but you try to fetch `/api/favorites/alltimefavorites/allfavorites`.

The same issue exists in your `updateFavorites` function that lives in `lib/updateFavorites`.

I hope this already does the trick!
Let me know otherwise :)

**Reply by t0bbl** [reply]
ok thats something to look at. i changed these, commited and pushed them locally but they didnt change in the PR. 
any tips how to force the new folder structure in to the pr?

**Reply by t0bbl** [reply]
ok thank you. it works now.

i changed the folder names to "folderXY1", git added,commited,pushed.
i changed the folder names back to "folderXY", git added, commited, pushed ...
its not stupid if it works!?
